CAIPs/caip-ipld_anchor_event.md

```diff
@@ -0,0 +1,150 @@
+
+If there is `chainID` parameter, we expect the `txId` to be an IPLD link to the transaction encoded using the related blockchain IPLD codec.
+
+```tsx
+type AnchorEvent struct {
+  root Link
+	chainID String 
+	txId Link
+}
+```
+
+Example 2
+```tsx
+{
+  root: CID(bafyreiaxdhr5jbabn7enmbb5wpmnm3dwpncbqjcnoneoj22f3sii5vfg74)
+	chainID: "eip155:1"
+	txId: CID(bagjqcgzanbud4sqdsywfp2mckuj57qsffsovgyjhh7sxebkqwr335hzy2zbq)
+}
+```
+
+If there is not a `chainID` parameter, we expect `txId` to be a transaction id string.
+
+```tsx
+type AnchorEvent struct {
+  root Link
+	txId String
+}
+```
+
+Example 3
+```tsx
+{
+  root: CID(bafyreiaxdhr5jbabn7enmbb5wpmnm3dwpncbqjcnoneoj22f3sii5vfg74)
+	txId: "eip155:1/txid:0x58be3f7b9d508d12b0c1cf2015a54169598210f1b40bf645777bcaf9cae656e8"
+}
+```
+
+Blockchain transaction resolution, parsing, and verification can later be defined by each blockchain CAIP-2 namespace. Implementations will vary and may even vary for the same chain. Optional parameters can be included here to indicate the methods required. For now transaction resolution, parsing, and verification is out of scope for this specification. 
+
+### IPLD Based Merkle Proofs
+
+A Merkle proof allows verification that a given piece of data or “leaf” was included in the set of data for the given Merkle Tree. A  typical proof includes a Merkle root of the tree and a subset list of tree node hashes that allow verification by reconstructing the Merkle root starting from the data or "leaf" you are interested in.
+
+In IPLD a Merkle root (CID) and a “path” are provided that allows you to traverse the tree from the root to the piece of data your interested in. Paths are strings and refer to [Pathing in IPLD](https://ipld.io/docs/data-model/pathing/). Following an IPLD path, and ultimately resolving the data or "leaf" you are interested in from the root, is equivalent to a standard proof. 
+
+As defined above, an IPLD Merkle tree node includes a list pair. List traversal in IPLD is defined by index, for example 0 referencing the first item, 1 referencing the second. An example IPLD path through a tree could look as such `0/0/1/0/1/1`
+
+### IPLD Anchor Proof
+
+The combination of an IPLD based Merkle proof and anchor event now make up the entirety of an anchor proof for a given piece of data. 
+
+```tsx
+type AnchorProof struct {
+  data Link
+	proof Link
+	path String 
+}
+```
+
+Example 4
+```tsx
+{
+  data: CID(bagcqcerav5zyrlelnnx24blbgob6stz6y4oiwenn3hodbmyf7bnnguh6gdeq)
+  proof: CID(bafyreietaedczjxh7omxfjm3oj55nbmxaga3himruubppgj3hp6dayonqq)
+	path: "0/0/1/0/1/1"
```

Comment:
In theory the `path` doesn't need to be though a binary merkle tree but any sort of IPLD graph. I think this could be clarified!

Reply:
the path is part of merkle tree proof, i dont imagine it is anything but an IPLD merkle tree, it just doesnt have to be a binary tree, could have "3/5/1/0" etc but still expect tree nodes to be lists of links 

in what case would it just be a general ipld graph?

Reply:
Yes, it will need to be a merkle tree. There's nothing forcing it to use arrays for the nodes though. So the path could be whatever essentially, e.g. `my/path/left/right`

Reply:
yeah definitely possible, and since defined with ipld paths it doesnt really need to follow particular format, can be verified 

but it seemed beneficial to define it more strictly here and define an ipld merkle tree for anchors, rather than leave entirely open ended,. not seeing benefits to leaving open

could add note/example for it being general, but think I would keep the ipld merkle tree construction in this spec as suggested format at least

Reply:
Sounds good.